sklearn/metrics/_ranking.py

```diff
@@ -112,6 +112,15 @@ def auc(x, y):
     return area
 
 
+@validate_params(
+    {
+        "y_true": ["array-like"],
+        "y_score": ["array-like"],
+        "average": [StrOptions({"micro", "samples", "weighted", "macro"})],
+        "pos_label": [Integral, str, 1],
```

Comment:
1 is already an `Integral`. However, I don't think that we should be limited to those 2 types since we could have also float classes and boolean. The following seems more appropriate.

```suggestions
        "pos_label": [Real, str, "boolean"],
```